exchanges/Saturn.js

```diff
@@ -0,0 +1,92 @@
+const rp = require('request-promise')
+const { SATURN_URL } = require('../constants.js')
+const OrderBookExchange = require('./OrderBookExchange.js')
+
+const etherAddress = '0x0000000000000000000000000000000000000000'
+
+module.exports = class SaturnNetwork extends OrderBookExchange {
+  constructor(chain = 'eth') {
+    super()
+    this.currenciesUrl = `${SATURN_URL}/dashboard/${chain}.json`
+    this.chain = chain
+    this.name = 'Saturn Network'
+  }
+
+  async _getCurrencies() {
+    try {
+      const config = {
+        timeout: 5000,
+        uri: this.currenciesUrl,
+        method: 'GET',
+        json: true,
+      }
+      const data = await rp(config)
+      return data.sort((a, b) => parseFloat(b.token.liquidity_depth.ether) - parseFloat(a.token.liquidity_depth.ether))
+    } catch (error) {
+      throw new Error(`error fetching data from ${this.name}: ${error}`)
+    }
+  }
+
+  async _getTokenAddress(symbol) {
+    const allTokens = await this._getCurrencies()
+    const tokenObj = allTokens.find(x => x.token.symbol === symbol)
+    return tokenObj.token.address
+  }
+
+  async _getRawOrderBook(symbol) {
+    try {
+      const tokenAddress = await this._getTokenAddress(symbol)
+      const url = `${SATURN_URL}/orders/${this.chain}/${tokenAddress}/${etherAddress}/all.json`
+      const config = {
+        timeout: 10000,
+        uri: url,
+        method: 'GET',
+        json: true,
+      }
+      const data = await rp(config)
+      return data
+    } catch (error) {
+      throw new Error(`error fetching sell rate from ${this.name}: ${error}`)
```

Comment:
a more appropriate error message here would be "error fetching order book data" instead of "error fetching sell rate"